scripts/workflow_validation_test_support.py

```
@@ -0,0 +1,217 @@
+)
+
+
+def load_workflow(path: Path) -> dict[str, object]:
+    """Parse one workflow and require a string-keyed mapping root."""
+    loader = WorkflowLoader(path.read_text(encoding="utf-8"))
+    try:
+        document = loader.get_single_data()
+    finally:
+        loader.dispose()
+    if not isinstance(document, dict):
+        raise AssertionError(f"{path} must parse to a mapping")
```

Ruff (0.16.3)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)